L59-core-allow-cppstdlib.md

```diff
@@ -0,0 +1,97 @@
+Allow C++ standard library in gRPC Core Library
+----
+* Author(s): veblush
+* Approver: a11r
+* Status: Draft
+* Implemented in: n/a
+* Last updated: August 19, 2019
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/umQyyXmNr2w
+
+## Abstract
+
+Allow C++ Standard Library to be used in the gRPC Core library.
+
+## Background
+
+Since C++ was allowed to write gRPC, more and more code has been written in C++.
+But most features from C++ standard libraries were not allowed to use because
+we didn't want to introduce new dependency against C++ standard library.
+This gRFC proposes allowing C++ standard library to be used within the core
+library, but continues to restrict the public interface to C89.
+
+### Related Proposals:
+
+* [Allow C++ in gRPC Core Library](L6-core-allow-cpp.md)
+
+## Proposal
+
+Allow C++11 standard library usage in gRPC Core. The previous C++ usage rule
+will be valid except the restriction on C++ features and header-only library.
+
+- All C++11 features are allowed to use. Followings were not possible to use
+  but it becomes availabie with this.
+  - `new` and `delete`
+    - Even though `new` and `delete` now become possible to use, it doesn't
+      necessarilly mean that all code should use.
+      For the place where `gpr_malloc` and `gpr_free` should be used
+      intentionally, `grpc_core::New` and `grpc_core::Delete` need to be
+      used.
+  - Pure virtual functions.
+    Regular form `= 0` can be used instead of `GRPC_ABSTRACT`
+
+- All features from C++11 standard library are allowed to use.
+  Previosuly only header-only features were allowed to use.
+
+There are caveats since gRPC wrapped library are being distributed as a binary
```

Comment:
we should describe the full story behind the caveat.
- wrapped languages' packages contain binary pre-compiled version of core
- packages need to be installable on a wide range of platforms without the user needing to install any extra packages (e.g. apt-get install some libraries).
- we must only use a libstdc++ version that's available on all of the platforms we supporting (which happens to be manylinux1?  which libstdc++ version is that?).
-